src/plugins/validation/oas3/semantic-validators/security-definitions-ibm.js

```diff
@@ -0,0 +1,175 @@
+        path
+      });
+    } else {
+      //apiKey validation
+      if (type === API_KEY) {
+        const authIn = security.in;
+        if (!authIn || !['query', 'header', 'cookie'].includes(authIn)) {
+          errors.push({
+            message:
+              "apiKey authorization must have required 'in' property, valid values are 'query' or 'header' or 'cookie'.",
+            path
+          });
+        }
+        if (!security.name) {
+          errors.push({
+            message:
+              "apiKey authorization must have required 'name' string property. The name of the header or query parameter to be used.",
+            path
+          });
+        }
+      }
+      // oauth2 validation
+      else if (type === OAUTH2) {
+        const flows = security.flows;
+        const tokenUrl = security.tokenUrl;
+
+        if (!flows) {
+          errors.push({
+            message:
+              "oauth2 authorization must have required 'flows' parameter",
+            path
+          });
+        } else if (
+          flows === AUTHORIZATION_CODE ||
+          flows === PASSWORD ||
+          flows === CLIENT_CREDENTIALS
+        ) {
+          if (!tokenUrl) {
+            errors.push({
+              message:
+                "oauth2 authorization authorizationCode flow must have required 'tokenUrl' string parameter if type is `authorizationCode`, `password`, `clientCredentials`.",
+              path
+            });
+          }
+        } else if (
+          !flows.implicit &&
+          !flows.authorizationCode &&
+          !flows.password &&
+          !flows.clientCredentials
+        ) {
+          errors.push({
+            message:
+              "oauth2 authorization `flows` must have one of the following paramaters: 'implicit', 'password', 'clientCredentials' or 'authorizationCode'",
+            path
+          });
+        } else if (flows.implicit) {
+          const authorizationUrl = flows.implicit.authorizationUrl;
+          if (!authorizationUrl) {
+            errors.push({
+              message:
+                "oauth2 authorization implicit flow must have required 'authorizationUrl' parameter if type is `implicit`.",
+              path
+            });
+          } else if (!flows.implicit.scopes) {
+            errors.push({
+              message:
+                "oauth2 authorization implicit flow must have required 'scopes' property.",
+              path
+            });
+          }
+        } else if (flows.authorizationCode) {
+          const authorizationUrl = flows.authorizationUrl;
+          if (!authorizationUrl) {
+            errors.push({
+              message:
+                "oauth2 authorization implicit flow must have required 'authorizationUrl' parameter if type is `implicit` or `authorizationCode`.",
+              path
+            });
+          }
+        } else if (flows.password) {
+          const tokenURL = flows.tokenURL;
```

Comment:
The `tokenURL` property actually lives _inside_ the `flows.password` object ... not directly in `flows`.